lib/config_validator.rb

```diff
@@ -1,24 +1,69 @@
 class ConfigValidator
-  def validate(env = ENV)
-    bad_keys = keys_with_bad_values(env, candidate_keys(env))
+  include Pii::Encodable
+
+  def initialize(env = ENV)
+    @env = env
+    @candidate_keys = keys_without_figaro_prefix
+  end
+
+  def validate
+    bad_keys = keys_with_bad_boolean_values + keys_with_bad_base64_encoding
     return unless bad_keys.any?
     raise warning(bad_keys).tr("\\\n", ' ')
   end
 
   private
 
-  def candidate_keys(env)
+  attr_reader :env, :candidate_keys
+
+  def keys_without_figaro_prefix
     env.keys.delete_if { |key| key.starts_with?(Figaro::Application::FIGARO_ENV_PREFIX) }
   end
 
-  def keys_with_bad_values(env, keys)
-    keys.keep_if { |key| %w(yes no).include?(env[key]) }
+  def keys_with_bad_boolean_values
+    candidate_keys.select { |key| %w(yes no).include?(env[key]) }
+  end
+
+  def keys_with_bad_base64_encoding
+    candidate_keys.select { |key| bad_base64_encoding?(key) }
+  end
+
+  def bad_base64_encoding?(key)
+    requires_base64_encoding?(key) && !valid_base64_encoding?(env[key])
+  end
+
+  def requires_base64_encoding?(key)
+    %w(attribute_encryption_key email_encryption_key).include?(key)
```

Comment:
how about suffixing with `_base64` or `_b64`, and checking for that? would also make processing of those keys a little more semantic.